### ISSUE TITLE
Dynamic API Path configuration

### ISSUE THREAD

**Issue by prophet1906** [chgeo/cds-swagger-ui-express#151]
The URL is usually dynamic when deploying the frontend to the HTML5 repo with swagger regex added as part of xs-app.json for serving swagger from the backend. 

Example - `abc.com/1234-appid/api-docs/...`, here swagger will load properly but API calls will fail because they will be sent to `abc.com/api-docs/...`.

I think hardcoding it for each landscape(dev/test/prod) is not a good idea.

It would be cool if default apiPath [here](https://github.com/chgeo/cds-swagger-ui-express/blob/main/lib/middleware.js#L17) was `/` instead of an empty string and the join [here](https://github.com/chgeo/cds-swagger-ui-express/blob/main/lib/middleware.js#L47) didn't include first `/`. This would allow users to configure the apiPath without the prefix `/` and should ideally allow dynamic basePath for different landscapes.

This will ensure URLs from different landscapes like`abc.com/1234-appid/api-docs/...`, `abc.com/5678-appid/api-docs/...`, `abc.com/9876-appid/api-docs/...`, will all work without any config changes if we do not add prefix `/` in apiPath.

The suggested changes won't break any existing app with default config while providing more control over configuration.

**Comment by chgeo** [comment]
Thanks for the input.
Can you check PR https://github.com/chgeo/cds-swagger-ui-express/pull/152 whether this what you are requesting?

**Comment by prophet1906** [comment]
Nope, I wanted the **1234-appid/9876-appid** to be configurable from env or maybe autodetected if possible.

**Comment by prophet1906** [comment]
The changes in PR #152 break existing code with simple version upgrade.

**Comment by chgeo** [comment]
@prophet1906 please provide a small reproducible example repo on what got broken.
